python/models.py

```diff
@@ -50,7 +88,9 @@ def __init__(
     ) -> None:
         self.tokenizer = AutoTokenizer.from_pretrained(name)
         if device == "auto":
-            device = get_cuda_if_available()
+            device = get_best_device_available()
+        elif device == "rocm":
+            device = get_rocm_if_available()
```

Comment:
Here is one more I missed. Consider adding `elif device == "cuda"`.

Reply:
Since there are 3 of such duplicate blocks, just extract a function then, possibly called `get_device` as I can think of.